pyat/at/physics/radiation.py

```diff
@@ -109,6 +116,25 @@ def propag(m, cumb, orbit6):
     nelems = len(ring)
     uint32refs = uint32_refpts(refpts, nelems)
     allrefs = uint32_refpts(range(nelems + 1), nelems)
+    if energy is None:
+        if isinstance(ring, at.lattice.Lattice):
+            energy = ring.energy
+        else:
+            rf_energies = []
```

Comment:
The strategy to get the energy should be the same as for creating a Lattice object. For instance the priority order could be:
- RingParam elements
- energy of RF cavities (I didn't think of that, but it's nice)
- energy attribute of any other element

And it would be even better to factorise the code with `Lattice.__init__`

Reply:
I will make a `find_ring_energy` function in `lattice.utils` then.